src/api/java/com/minecolonies/api/colony/managers/interfaces/IColonyPackageManager.java

```diff
@@ -35,50 +35,61 @@
 
     /**
      * Update the colony view.
-     * @param oldSubscribers the old subs.
-     * @param hasNewSubscribers if there are new subs.
      */
-    void sendColonyViewPackets(@NotNull final Set<EntityPlayerMP> oldSubscribers, final boolean hasNewSubscribers);
+    void sendColonyViewPackets();
 
     /**
      * Sends packages to update the permissions.
-     *
-     * @param oldSubscribers    the existing subscribers.
-     * @param hasNewSubscribers the new subscribers.
      */
-    void sendPermissionsPackets(@NotNull final Set<EntityPlayerMP> oldSubscribers, final boolean hasNewSubscribers);
+    void sendPermissionsPackets();
 
 
     /**
      * Sends packages to update the workOrders.
-     *
-     * @param oldSubscribers    the existing subscribers.
-     * @param hasNewSubscribers the new subscribers.
      */
-    void sendWorkOrderPackets(@NotNull final Set<EntityPlayerMP> oldSubscribers, final boolean hasNewSubscribers);
+    void sendWorkOrderPackets();
 
 
     /**
      * Sends packages to update the schematics.
-     *
-     * @param hasNewSubscribers the new subscribers.
      */
-    void sendSchematicsPackets(final boolean hasNewSubscribers);
+    void sendSchematicsPackets();
 
     /**
      * Mark the package manager dirty.
      */
     void setDirty();
 
     /**
-     * Add a new subsriber to the colony.
+     * Add a new subscriber to the colony.
      * @param subscriber the subscriber to add.
      */
     void addSubscribers(@NotNull final EntityPlayerMP subscriber);
 
     /**
-     * Remove an old subsriber from the colony.
+     * Adds a new global subscriber to the colony.
```

Comment:
More documentation is needed:
  * What is a Global Subscriber?
  * What are they user for?

Reply:
Also:
  * Why are they needed, what can your new system not do, that you need this kind of brute force override for, that the current system can do?

Reply:
you're right I should describe the subscribers better. In this new system they're divided into two groups:
- close subscribers, those are within the colony and receive all the updates/views
  - Those are populated on chunk enter
- global subscribers, those are used for out of colony functionality like keeping the colony active when chunks are loaded, or receiving important messages of stuff happening within the colony like a raider event.
  - Those are populated upon login/promote

Reply:
The new system is based on the cunk enter/login events instead of constantly querying all online players as the old system did

Reply:
Okey but then this needs to be better documented and then name `global subscriber` is completely misleading since it implies that you are a subscriber of colony data, regardless of where you are or not.